### ISSUE TITLE
LaTex tables

### ISSUE THREAD

**Issue by kellijohnson-NOAA** [klpn/accpdf#1]
Thank you for your hard work and for posting this code. It has been instrumental in the creation of accessible documents. I am having trouble with the rendering of tables though if they are not in markdown format. For example, I can add the following to your markdown script in tagged-article, which is just the LaTex version of the table already in the .md file,
```
\begin{longtable}[]{@{}lll@{}}
\caption{Lifetime probabilty of cancer incidence and mortality for
Swedish females and males 2014.}\tabularnewline
\toprule
Measure & Female & Male\tabularnewline
\midrule
\endfirsthead
\toprule
Measure & Female & Male\tabularnewline
\midrule
\endhead
Incidence & 45.7 & 49.3\tabularnewline
Mortality & 22.3 & 26.2\tabularnewline
\bottomrule
\end{longtable}
```
and an additional table is generated in the pad but it is not tagged. I am not certain what is going on. Any chance you can help? Thank you in advance. I have many tables that are results from xtable in R, where the added functionality of LaTex is needed compared to simple markdown tables.

**Comment by klpn** [comment]
As noted in the README, the package, and the tagpdf package, which it uses, are both experimental.  The problem you mention seems to arise because the second table is interpreted as raw LaTeX by the markdown reader, while the package uses a Lua filter to add tags to Table elements in the Pandoc AST.

**Comment by kellijohnson-NOAA** [comment]
Thank you for the summary of the issue and for making the package in general. I realize it is experimental and I am happy to help with development if needed but I haven't been very successful with respect to this table issue. Do you know of any way that I can facilitate pandoc recognizing it as a table so the appropriate tags are added? I am trying to go from .Rmd to .md to .tex to .pdf via bookdown, a package in R, and writing markdown style tables just isn't an option. If you don't have any ideas on how to make LaTex tables work with the filter maybe you can point me towards some Lua documentation that was helpful for you when you built this package? Thank you.

**Comment by klpn** [comment]
The construction of Pandoc Lua filters is described in [the Pandoc documentation](https://pandoc.org/lua-filters.html). One way might be to write a `RawBlock` function which identifies the relevant LaTeX commands for tables, like in the TikZ example on the page, and then do some substitutions inside the block to insert tagging commands.

**Comment by kellijohnson-NOAA** [comment]
Great, I will look at that example. And thanks again for the quick responses, I really appreciate it.

**Comment by kellijohnson-NOAA** [comment]
Thank you for your help. I was able to first print the AST structure to the screen, so see how the latex tables were being rendered, using json. That confirmed that the latex was a "RawBlock" with a $format of "latex". I then was able to follow the referenced examples and use an if then statement
```
function RawBlock(el)
  if el.format:match "html" then
    return pandoc.RawBlock("html", el)
  else
    local oldlatex = rb.text
    local out = pandoc.pip('sed', <sed calls go here>, oldlatex)
    return pandoc.RawBlock("latex", out)
end
end
```
where users would have to replace `<sed calls go here>` with their own text minus the brackets. Mine obviously pertained to finding instances of begin{longtable} and end{longtable} where I added lines before and after specifying the Table tag that I needed. My ability to use sed is very immature, thus I am not providing my explicit calls b/c I don't want to goof anyone up in the future that uses this issue as a resource. Feel free to contact me though if you want the exact code that I used. In the end, I was able to make non markdown tables inside of a markdown file 508 compliant using this filter when building a pdf. Thank you @klpn for your help and for creating this repository. I can't say enough regarding how much it helped me.
